### PR TITLE
feat: add ratelimit to response

### PR DESCRIPTION
**Pull request by peterdeme**
- Add rate limit to response body

Adding @madsroskar since he has some PHP experience.

### CODE REVIEW

File: lib/GetStream/StreamChat/StreamResponse.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * A class that extends ArrayObject and additionally contains response metadata
+ * such as rate limits.
+ */
+class StreamResponse extends \ArrayObject
+{
+    /**
+     * @var string[]|null
+     */
+    private $rateLimits;
```
Comment (AnatolyRugalev) [inline lib/GetStream/StreamChat/StreamResponse.php:16]: Can we make a class with 3 properties for rate limits?

**Review by AnatolyRugalev — APPROVED** [review]
Looks good overall, one suggestion attached

**Review by ferhatelmas — APPROVED** [review]

**Review by madsroskar — APPROVED** [review]

File: lib/GetStream/StreamChat/StreamResponse.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        parent::__construct($array);
+
+        if ($response->hasHeader("x-ratelimit-limit")) {
+            $this->rateLimits = new StreamRateLimit(
+                $response->getHeader("x-ratelimit-limit")[0],
+                $response->getHeader("x-ratelimit-remaining")[0],
+                $response->getHeader("x-ratelimit-reset")[0]
+            );
+        }
```
Comment (AnatolyRugalev) [inline lib/GetStream/StreamChat/StreamResponse.php:37]: There's a small chance that one of these headers will be missing. We should ideally check presence of all of them

Comment (peterdeme) [reply lib/GetStream/StreamChat/StreamResponse.php]: @AnatolyRugalev ✅

**Review by AnatolyRugalev — APPROVED** [review]
Perfect, thanks

**Pull request merged**